cloudferrylib/os/actions/check_needed_compute_resources.py

```diff
@@ -0,0 +1,62 @@
+# Copyright (c) 2015 Mirantis Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and#
+# limitations under the License.
+
+import collections
+
+from cloudferrylib.base.action import action
+from cloudferrylib.utils import utils as utl
+
+
+class CheckNeededComputeResources(action.Action):
+    def run(self, **kwargs):
+        info = kwargs['info_iter']
+        objs = info[utl.INSTANCES_TYPE]
+        cnt_map = collections.defaultdict(int)
+        for instance in objs.values():
+            cnt_map[instance['instance']['flavor_id']] += 1
+        needed_cpu = 0
+        needed_ram = 0
+        needed_hdd = 0
+        src_nova = self.src_cloud.resources[utl.COMPUTE_RESOURCE]
+        for id, count in cnt_map.get_map().items():
+            flavor = src_nova.get_flavor_from_id(id)
+            needed_cpu += flavor.vcpus * count
+            needed_hdd += (flavor.disk + flavor.ephemeral) * count
+            if flavor.swap:  # if flavor not specified '' is here
+                needed_hdd += float(flavor.swap) * count / 1024
+                # flavor in Mb instead of disk
+            needed_ram += flavor.ram * count
+        dst_nova = self.dst_cloud.resources[utl.COMPUTE_RESOURCE]
+        hypervisor_statistics = dst_nova.get_hypervisor_statistics()
+        if hypervisor_statistics.vcpus - hypervisor_statistics.vcpus_used < \
```

Comment:
This estimate might be incorrect in cases when:
 - CPU overcommit ratio is different from 1:1, for example by default nova sets it to 16:1 (1 physical CPU == 16 vCPUs)
 - In case physical CPU supports hyperthreading (which is pretty common), the overcommit ratio should be 1.3:1 (or 2:1 according to [openstsack docs](http://docs.openstack.org/openstack-ops/content/compute_nodes.html)).

In order to cover these cases we should:

1. Pick the `cpu_allocation_ratio`, `ram_allocation_ratio` and `disk_allocation_ratio` from `nova.conf`.
2. Multiply *actual resources* by the corresponding factor, and compare those against `needed_<resource>`